app/controllers/projects/new_tree_controller.rb

```diff
@@ -6,15 +6,65 @@ def show
   end
 
   def update
-    file_path = File.join(@path, File.basename(params[:file_name]))
-    result = Files::CreateService.new(@project, current_user, params, @ref, file_path).execute
-
+	flag=0
+	#replace & upload 
+	if params[:file_upload] != nil
+		#get the name of the upload file
+		file_na=params[:file_upload].original_filename
+		file_path=nil
+		
+	    if file_na != nil
+			file_list = tree.entries.select(&:file?)
+			dir_list = tree.entries.select(&:dir?)
+			
+			dir_list.each do |dir|		#check whether current path is sub-repository
+				if dir.name != nil
+					flag=1
```

Comment:
Use 2 (not 1) spaces for indentation.<br>Surrounding space missing for operator '='.<br>Tab detected.